pi/test_heartbeat.py

```diff
@@ -42,18 +43,38 @@ def test_log_last_seen_when_darkstat_html_contains_flic_mac_address(self):
 class Test__parse_flic_ip_from_darkstat_html(object):
 
     def test_valid_input(self):
-        with open(os.path.dirname(__file__) + '/sample_darkstat_html/59_secs.html', 'r') as html_file:
+        with open(os.path.dirname(__file__) + '/test_files/sample_darkstat_html/59_secs.html', 'r') as html_file:
             html = html_file.read()
             assert heartbeat.parse_flic_ip_from_darkstat_html(html, '00:00:00:00:00:02') == '192.168.8.114'
 
     def test_log_ip_when_darkstat_html_contains_flic_mac_address(self):
-        file_path = os.path.dirname(__file__) + '/sample_darkstat_html/log_ip.html'
+        file_path = os.path.dirname(__file__) + '/test_files/sample_darkstat_html/log_ip.html'
         with open(file_path, 'r') as html_file, unittest.mock.patch('heartbeat.logging') as mock_logging:
             html = html_file.read()
             heartbeat.parse_flic_ip_from_darkstat_html(html, '00:00:00:00:00:02')
 
             mock_logging.info.assert_called_once_with('darkstat html contained flic IPv4 address:  <td><a href="./192.168.8.114/">192.168.8.114</a></td>  <td>flic</td>  <td><tt>00:00:00:00:00:02</tt></td>  <td class="num">1 hr, 3 mins, 26 secs</td></tr>')
 
+class Test__parse_link_quality_from_iwconfig_output(object):
```

Comment:
Thanks for writing tests :heart: 

Might be worth checking to see whether the string is logged, not just that the value is calculated correctly though.

Reply:
The issue with this is that if `iwconfig` isn't present on the test machine, the test will fail. I guess the test could check for the presence of `iwconfig` and then not run? But I feel like this will often be the case - ie. I doubt the Travis virtual boxes have a wlan0 interface, `iwconfig` doesn't work on my WSL dev environment, etc.

Reply:
I guess I could also mock the `subprocess.check_output` call, but at that point the additional tests would basically just be 'does the logging library work'

Reply:
What do you think about all this @trisapeace?

Reply:
What do you think of the `mock_logging.info.assert_called_once_with(...)` construct used in the test directly above you new ones?

Reply:
ok I've updated the implementation so that this test structure is possible, and changed the tests as well.